src/player.h

```diff
@@ -73,6 +73,18 @@ struct special_attack {
     };
 };
 
+class player_morale;
+class player_morale_ptr : public std::unique_ptr<player_morale> {
```

Comment:
Would be better if you reused the template instead. This one works, but now we have a template that does thing and something that could easily be an instance of said template doing the same thing.